### PR TITLE
Add help to scripts and add groups to tests

### PR DESCRIPTION
**Pull request by M-Aarsman**

### CODE REVIEW

File: scripts/cloneAndAddHook.sh [hunk 1]
```diff
@@ -15,9 +15,42 @@
...
+echo "* It is dependent on pre-commitHookMaven.sh script"
+echo
+echo "[Usage]"
+echo -e "\t ./cloneAndHook.sh [your_dir_name]"
+echo -e "\t\t [your_dir_name] - dir to which you want to clone"
+}
+
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
Comment (robwrobel) [inline scripts/cloneAndAddHook.sh]: exit 0?

File: scripts/cloneAndAddHook.sh [hunk 2]
```diff
@@ -15,9 +15,42 @@
...
+fi
 cd $1
 echo "copying pre-commit hook"
-chmod u+x scripts/pre-commitHookMaven.sh
-cp scripts/pre-commitHookMaven.sh .git/hooks/pre-commit
-echo "hook added"
\ No newline at end of file
+if ! chmod u+x scripts/pre-commitHookMaven.sh;
```
Comment (robwrobel) [inline scripts/cloneAndAddHook.sh]: I think pre-commitHookMaven.sh has already execute permission so chmod is not needed

File: scripts/countTests.sh [hunk 3]
```diff
@@ -10,6 +10,20 @@
...
+echo
+echo "It runs mvn test, and basing on results it counts number of unit tests in project."
+echo
+echo "[Usage]"
+echo -e "\t ./countTests.sh\n"
+}
+
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
Comment (robwrobel) [inline scripts/countTests.sh]: exit 0

File: scripts/countTests.sh [hunk 4]
```diff
@@ -10,6 +10,20 @@
...
+fi
+
```
Comment (robwrobel) [inline scripts/countTests.sh]: missing cd ..   . mvn test will not run from scripts directory

File: scripts/deployserver.sh [hunk 5]
```diff
@@ -16,10 +16,43 @@
...
+echo -e " \t 1) The assumption is that the script is placed within ./scripts/ directory."
+echo -e " \t 2) JAR file name has to be maintained manually."
+echo
+echo "[Usage]"
+echo -e "\t ./deployserver.sh\n"
+}
+
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
Comment (robwrobel) [inline scripts/deployserver.sh]: exit 0

File: scripts/deployserver.sh [hunk 6]
```diff
@@ -16,10 +16,43 @@
...
+then
+	helpEcho
+	exit 1;
+fi
+if ! mvn clean -q;
+then
+	helpEcho
+	exit 1;
+fi
+if ! mvn install -q;
```
Comment (robwrobel) [inline scripts/deployserver.sh]: when I run this script on clean system, I got following error:
./deployserver.sh
[ERROR] Failed to execute goal on project server: Could not resolve dependencies for project com.epam.ships:server:jar:1.0-SNAPSHOT: Could not find artifact com.epam.ships:shared:jar:1.0-SNAPSHOT -> [Help 1]

File: scripts/deployserverandclients.sh [hunk 7]
```diff
@@ -16,13 +16,54 @@
...
+echo -e " \t 1) The assumption is that the script is placed within ./scripts/ directory."
+echo -e " \t 2) JAR file name has to be maintained manually."
+echo
+echo "[Usage]"
+echo -e "\t ./deployserverandclients.sh\n"
+}
+
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
Comment (robwrobel) [inline scripts/deployserverandclients.sh]: exit 0

File: scripts/generateLocalSite.sh [hunk 8]
```diff
@@ -17,15 +17,50 @@
...
+echo -e " \t * It takes a few minutes to execute"
+echo -e " \t * The assumption is that the repo is up-to-date and there aren't uncommitted changes on current branch"
+echo
+echo "[Usage]"
+echo -e "\t ./generateLocalSite.sh\n"
+}
+
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
Comment (robwrobel) [inline scripts/generateLocalSite.sh]: exit 0

File: scripts/summarize.sh [hunk 9]
```diff
@@ -22,31 +22,80 @@
...
+echo -e " \t 1) Make sure the branch you're on has no uncommitted changes. Auto-commit wasn't introduced for safety reasons."
+echo -e " 2) The assumption is that the script is placed within ./scripts/ directory."
+echo
+echo "[Usage]"
+echo -e "\t ./summarize.sh\n"
+}
+
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
Comment (robwrobel) [inline scripts/summarize.sh]: exit 0

File: scripts/updateRemoteSite.sh [hunk 10]
```diff
@@ -18,38 +18,137 @@
...
+echo -e " \t * It requires a user to provide his credentials at some stage (when performing git push)"
+echo 
+echo "[Usage]"
+echo -e "\t ./updateRemoteSite.sh\n"
+}
 
-mkdir ../../tempUpdateRemoteSite
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
Comment (robwrobel) [inline scripts/updateRemoteSite.sh]: exit 0

File: scripts/clone.sh [hunk 11]
```diff
@@ -12,8 +12,36 @@
...
+
+if ! git clone https://github.com/korotkevics/ships $1 --quiet;
+then
+	helpEcho
+	exit 1;
+fi
+
 cd $1
-git checkout master --quiet
+if ! git checkout master --quiet;
```
[the remark this answers is not in the document]
Comment (korotkevics) [reply scripts/clone.sh]: It all started with demo when it was required to clone a repo and show stuff on master (and our default branch has been int..) :)

File: scripts/countTests.sh (hunk 4 above)
Comment (korotkevics) [reply scripts/countTests.sh]: it is a part of another script which does that, should we mention it somewhere?

**Comment by korotkevics** [comment]
@M-Aarsman I'd prefer if you respond to remaining comments because they're closely related to your changes, and when it comes to quiet commands I'd implement logging their output with another pull later, it is a nice advice from @robwrobel thanks! I've created a card for that on Waffle.

**Comment by korotkevics** [comment]
@M-Aarsman I think I'll fix them myself to speed up:)

File: scripts/clone.sh [hunk 12]
```diff
@@ -12,8 +12,36 @@
...
+echo " 1) Copy a script (if not willing to alias it) into a directory"
+echo
+echo "[Usage]"
+echo -e "\t ./clone.sh [your_dir_name]"
+echo -e "\t\t [your_dir_name] - dir to which you want to clone"
+}
+
+if [[ $1 = "--help" ]];
+	then helpEcho
+	exit 1;
```
[the remark this answers is not in the document]
Comment (korotkevics) [reply scripts/clone.sh]: Done

File: scripts/cloneAndAddHook.sh (hunk 1 above)
Comment (korotkevics) [reply scripts/cloneAndAddHook.sh]: Done

File: scripts/countTests.sh (hunk 3 above)
Comment (korotkevics) [reply scripts/countTests.sh]: Done

File: scripts/cloneAndAddHook.sh (hunk 2 above)
Comment (korotkevics) [reply scripts/cloneAndAddHook.sh]: Removed

File: scripts/deployserver.sh (hunk 5 above)
Comment (korotkevics) [reply scripts/deployserver.sh]: Done

File: scripts/deployserverandclients.sh (hunk 7 above)
Comment (korotkevics) [reply scripts/deployserverandclients.sh]: Done

File: scripts/generateLocalSite.sh (hunk 8 above)
Comment (korotkevics) [reply scripts/generateLocalSite.sh]: Done

File: scripts/updateRemoteSite.sh (hunk 10 above)
Comment (korotkevics) [reply scripts/updateRemoteSite.sh]: Done

File: scripts/summarize.sh (hunk 9 above)
Comment (korotkevics) [reply scripts/summarize.sh]: Done

File: scripts/countTests.sh (hunk 4 above)
Comment (robwrobel) [reply scripts/countTests.sh]: maybe it is a good idea to write that this script cannot be run standalone

**Comment by korotkevics** [comment]
@robwrobel could you please review changes

**Pull request merged**